src/Observables.jl

```diff
@@ -18,20 +18,55 @@ end
 """
 Like a `Ref` but updates can be watched by adding a handler using `on`.
 """
-mutable struct Observable{T} <: AbstractObservable{T}
-    id::String
-    val::T
+struct Observable{T} <: AbstractObservable{T}
+    id::UInt64
+    val::RefValue{T}
     listeners::Vector
 end
-Observable{T}(val) where {T} = Observable{T}(newid(), val, Any[])
-Observable(val::T) where {T} = Observable{T}(val)
+
+const observable_names = Dict{UInt64, String}()
+
+
+"""
+    rename!(x::AbstractObservable, name)
+
+Renames an observable
+"""
+rename!(x::AbstractObservable, name) = observable_names[obsid(x)] = string(name)
+
+"""
+    name!(x::AbstractObservable, name)
+
+Optional naming infrastructure for observables.
+name!(observable, "the name") will associate `observable` with the `the name`,
+and can be retrieved via `name(observable)`. If an observable already has a name,
+nothing will be done! Call `rename!(x::AbstractObservable, name)`, if you want to
+change the name.
+"""
+function name!(x::AbstractObservable, name)
+    id = obsid(x)
+    if !haskey(observable_names, id)
+        observable_names[id] = string(name)
+    end
+end
+
+"""
+    name(x::Observable, default = "noname")
+if observable `x` is named, returns the name and otherwise returns `default`.
+"""
+function name(x::Observable, default = "noname")
+    get(observable_names, obsid(x), default)
+end
```

Comment:
TBH, I'm not sure this observable naming stuff really belongs in the package. The implementation is kind of inherently global and there's no way to clean up values from `observable_names` when an observable go out of scope. Coupled with the fact that it's trivial for a user to implement this themselves... perhaps best just to leave it out?

But maybe you have a compelling use case you could describe?

Reply:
I actually don't, just wanted to preserve the old behaviour somehow... Although unique ids substantially clash with namings ;)